composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -27,9 +27,33 @@ def set_batch_sequence_length(batch: Dict[str, Tensor], curr_seq_len: int, trunc
     .. note::
 
         The schedule for ``curr_seq_len`` over training time should be managed
-        out of this function.
+        outside of this function.
 
-    Example: Awaiting language model test fixtures.
+    .. note::
+
+        Variable input lengths can create CUDA OOM errors. To avoid this,
+        we follow `PyTorch notes <https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#pre-allocate-memory-in-case-of-variable-input-length>`_
+        and pre-allocate the memory with a blank forward and backward pass.
+
+    Example:
+    
+    .. code-block::
+
+        import composer.functional as cf
```

Comment:
Can this example go after the Returns? I know a few of our functions have it before right now but I'm going to move them all to after for consistency with all other python libraries I've seen.